js/util/combineCriteria.ts

```diff
@@ -0,0 +1,31 @@
+import * as Collections from 'typescript-collections';
+import { PropExtractorCriteria, PropExtractorConfig } from './extractProps';
+
+export const getCombinedCriteria = (
+  componentName: string,
+  configs: PropExtractorConfig[]
```

Comment:
Why not take an array of `PropExtractorCriteria`s instead?

Reply:
Could - and I think I had it that way at one point.  Not sure I can think of a good reason to change it, as long as the contract is solid.  It'd just be shifting the extraction logic around to other spots.

Reply:
The reason I'm thinking is that `componentName` is only passed in to get the `PropExtractorCriteria` from the `PropExtractorConfig`.  Additionally, at the call site, `classCriteria` is getting converted from a `PropExtractorCriteria` to a `PropExtractorConfig` for the sole purpose of matching this function's signature.  So `classCriteria` is getting converted to a config, then back to a criteria.

To me, this method should just take an array of criteria, merge them together, and return the result.

Reply:
done